File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def get_divisor_count(n):
    cnt = 0
    for i in range(1, int(n ** 0.5) + 1):
        if n % i == 0:
            cnt += 1
            if i * i != n:
                cnt += 1
    return cnt
            

def solution(number, limit, power):
    answer = 0
    for n in range(1, number + 1):
        cnt = get_divisor_count(n)
        if cnt <= limit:
            answer += cnt
        else:
            answer += power
    return answer

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        for (int n = 1; n < number + 1; n++) {
            int count = getDivisorCount(n);
            if (count <= limit) {
                answer += count;
            } else {
                answer += power;
            }
        }
        return answer;
    }
    
    public int getDivisorCount(int n) {
        int count = 0;
        for (int i = 1; i < (int) Math.sqrt(n) + 1; i++) {
            if (n % i == 0) {
                count += 1;
                if (i * i != n) {
                    count += 1;
                }
            }
        }
        return count;
    }
}